### PR TITLE
[Configuration] Fix error messages

### LINKED ISSUE #6291: [Configuration/ Projects and subprojects] Usability: Every time a project/subproject is edited/created the browser send the user to the first project/subproject.
**Is your feature request related to an existing module?**
Configuration Module --> projects and subprojects

**Describe the solution you'd like**
Every time a project/subproject is edited/created the browser send the user to the first project/subproject.

**Describe alternatives you've considered**
After saving (submitting the changes) the user should remain in the same project/subproject she/he edited allowing to review the changes made.

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

Fixes a bad error message when submitting a subproject change.

Also cleans up the code.

EDIT: This also resolves #6291. I disabled the refresh entirely because it doesn't seem to provide any advantage; the changes a user makes are already reflected back to them.

#### Testing instructions (if applicable)

1. Go to Configuration and update an existing subproject.
2. Modify recruitment target to a non-integer value or a number less than or equal to zero
3. Make sure you get an appropriate error message.

#### Link(s) to related issue(s)

* Resolves https://github.com/aces/Loris/issues/6290

### CODE REVIEW

File: modules/configuration/ajax/updateSubproject.php [hunk 1]
```diff
@@ -23,11 +18,19 @@
...
 
-$factory = NDB_Factory::singleton();
-$db      = $factory->database();
+$db = $factory->database();
 $SubprojectList = Utility::getSubprojectList();
-$recTarget      = empty($_POST['RecruitmentTarget'])
-    ? null : $_POST['RecruitmentTarget'];
+$recTarget      = $_POST['RecruitmentTarget'];
+
+//error_log(sprintf("Recruitment target: `%s`", $recTarget));
```
Comment (ridz1208) [inline modules/configuration/ajax/updateSubproject.php:25]: should be deleted

**Comment by ridz1208** [comment]
@racostas I'll let you test it see if it resolves your issues

**Comment by racostas** [comment]
@johnsaigle The error message was fixed!!

I have some doubts with regard to the way it solves #6291. Now that the page is not refreshing the user remains in the tab is was editing, but due to the content is 'not refreshed of course' the name of the tab for a new created subproject remains `New SubprojectID` even in the case the user navigate to other tab. This could be a bit confusing... I'm thinking also the case (not to probable, but possible) of a user creating multiple subprojects at the time. In other hand if the `Subproject Name` is changed for a new created project (yet tagged as `New SubprojectID` in the tabs) the behaviour I observed now is that a new project is created in place of modifying this one that the user is creating (the only way to see it manually refreshing the page) ... maybe this fix could take more design and time that the one we have available for the release... just thinking... we could leave it  for the next one as it not a critical one. The error message is OK now. 

image

**Comment by johnsaigle** [comment]
Thanks @racostas for the review. I'll take a look today and try to get the refresh to work as you originally described. Until then, or until we decide it's not needed, I'll leave this as Needs Work.

**Comment by johnsaigle** [comment]
@racostas I don't know enough JavaScript to know how to do this 🤷 

The behaviour of this page seems unusual also. E.g. navigating to `https://jsaigle-dev.loris.ca/configuration/subproject/#subproject5` does not actually load `#subproject5`. It will always go to the first page.

@maltheism Would you mind taking a quick look since you're more knowledgable about this sort of thing?

If it's not a quick fix, I think we should leave this for when configuration gets Reactified.

**Comment by maltheism** [comment]
Hey @johnsaigle, I propose this fix for now until the module is updated to react. 

Update the `successClosure` inside Loris/modules/configuration/js/project.js
```
var successClosure = function(i, form) {
        return function() {
          $(form.find(".saveStatus")).text("Successfully saved").css({ 'color': 'black'}).fadeIn(500).delay(1000).fadeOut(500);
          if (ProjectID === 'new') {
            setTimeout(function(){
              location.reload();
            }, 1000);
          } else {
            var projectDiv = document.getElementById(`#project${ProjectID}`);
            projectDiv.innerText = Name;
            var projectHeader = document.getElementById(`project${ProjectID}`);
            projectHeader.children[0].innerText = Name + projectHeader.children[0].innerText.substring(Name.length);
          }
        }
      }
```

Update the Loris/modules/configuration/templates/form_project.tpl to be:
```
<script language="javascript" src="{$baseurl}/configuration/js/project.js">
</script>
<p>Use this page to manage the configuration of existing projects, or to add a new one.</p>
<p>To configure study subprojects <a href="{$baseurl}/configuration/subproject/">click here</a>.</p>

<div class="col-md-3">
<ul class="nav nav-pills nav-stacked" role="tablist" data-tabs="tabs">
    <li class="active"><a id="#projectnew{$ProjectID}" href="#projectnew" data-toggle="tab" class="active">New ProjectID</a></li>
    {foreach from=$projects key=ProjectID item=project name=configContent}
    <li><a id="#project{$ProjectID}" href="#project{$ProjectID}" data-toggle="tab">{$project.Name}</a></li>
    {/foreach}
</ul>
</div>

<div class="col-md-7">
    <div class="tab-content">
    {foreach from=$projects key=ProjectID item=project name=tabContent}
    <div id="project{$ProjectID}" class="tab-pane">
        <h2>{$project.Name} (ProjectID: {$ProjectID})</h2>
        <br>
        <form class="form-horizontal" role="form" method="post" id="form{$ProjectID}">
            <fieldset>
                <input type="hidden" name="ProjectID" value="{$ProjectID}" class="ProjectID">
                    <div class="form-group">
                        <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'Full descriptive title of the project'}">
                            <label class="col-sm-12 control-label">Project Name</label>
                        </div>
                        <div class="col-sm-12 col-md-9">
                            <input class="form-control projectName" name="Name" value="{$project.Name}">
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'Short name of the project (4 characters or less)'}">
                            <label class="col-sm-12 control-label">Alias</label>
                        </div>
                        <div class="col-sm-12 col-md-9">
                            <input class="form-control projectAlias" name="Alias" value="{$project.Alias}">
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'The target number will be used to generate the recruitment progress bar on the dashboard'}">
                            <label class="col-sm-12 control-label">Recruitment Target</label>
                        </div>
                        <div class="col-sm-12 col-md-9">
                            <input class="form-control projectrecruitmentTarget" name="recruitmentTarget" value="{$project.recruitmentTarget}">
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'These subprojects will be automatically displayed for any candidate affiliated with this project at timepoint creation.'}">
                            <label class="col-sm-12 control-label">Affiliated Subprojects</label>
                        </div>
                        <div class="col-sm-12 col-md-9">
                            <select name="SubprojectIDs" class="form-control projectSubprojectIDs" multiple>
                              {foreach from=$subprojects key=SubprojectID item=subproject}
                                  {if $subproject|in_array:$project.projectSubprojects}
                                      <option value="{$SubprojectID}" selected>{$subproject}</option>
                                  {else}
                                      <option value="{$SubprojectID}">{$subproject}</option>
                                  {/if}
                              {/foreach}
                            </select>
                        </div>
                    </div>
                    <div class="form-group">
                        <div class="col-sm-offset-3 col-sm-9">
                            <button id="saveproject{$ProjectID}" class="btn btn-primary saveproject submit-area">Save</button>
                            <button class="btn btn-default submit-area" type="reset">Reset</button>
                            <label class="saveStatus"></label>
                        </div>
                    </div>

            </fieldset>
        </form>
    </div>
    {/foreach}
    <div id="projectnew" class="tab-pane active">
        <h2>New Project</h2>
        <br>
        <form class="form-horizontal" role="form" method="post" id="form{$ProjectID}">
            <fieldset>
                <input type="hidden" name="ProjectID" value="new" class="ProjectID">
                <div class="form-group">
                    <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'Full descriptive title of the project'}">
                        <label class="col-sm-12 control-label">Project Name</label>
                    </div>
                    <div class="col-sm-12 col-md-9">
                        <input class="form-control projectName" name="Name" placeholder="Please add a project title here" value="">
                    </div>
                </div>
                <div class="form-group">
                    <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'Short name of the project (4 characters or less)'}">
                        <label class="col-sm-12 control-label">Alias</label>
                    </div>
                    <div class="col-sm-12 col-md-9">
                        <input class="form-control projectAlias" name="Alias" placeholder="Please add an alias here" value="">
                    </div>
                </div>
                <div class="form-group">
                    <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'The target number will be used to generate the recruitment progress bar on the dashboard'}">
                        <label class="col-sm-12 control-label">Recruitment Target</label>
                    </div>
                    <div class="col-sm-12 col-md-9">
			            <input class="form-control projectrecruitmentTarget" name="recruitmentTarget" placeholder="Please add a recruitment target here" value="">
                    </div>
                </div>
                <div>
                  <div class="col-sm-12 col-md-3" data-toggle="tooltip" data-placement="right" title="{'These subprojects will be automatically displayed for any candidate affiliated with this project at timepoint creation.'}">
                    <label class="col-sm-12 control-label">Affiliated Subprojects</label>
                  </div>
                  <div class="col-sm-12 col-md-9">
                    <div class="col-sm-12 col-md-9">
                      <select name="SubprojectIDs" class="form-control projectSubprojectIDs" multiple>
                        {foreach from=$subprojects key=SubprojectID item=subproject}
                            <option value="{$SubprojectID}">{$subproject}</option>
                        {/foreach}
                      </select>
                    </div>
                </div>
                <div class="form-group">
                    <div class="col-sm-offset-3 col-sm-9">
                        <button id="saveprojectnew" class="btn btn-primary saveproject submit-area">Save</button>
                        <button class="btn btn-default submit-area" type="reset">Reset</button>
                        <label class="saveStatus"></label>
                    </div>
                </div>

            </fieldset>
        </form>
    </div>
</div>
```

**Comment by johnsaigle** [comment]
@maltheism thanks for checking it out and making a suggestion!

do you have this work saved on a branch? I'd like to check it out and apply it as a commit after testing if it works (rather than copy paste the above).

**Comment by maltheism** [comment]
Hey @johnsaigle https://github.com/maltheism/Loris/tree/suggestion_for_john

**Comment by ridz1208** [comment]
@maltheism what's the state of this ?

**Comment by maltheism** [comment]
@ridz1208 the previous comment I made has a branch of the code changes I suggested for @johnsaigle to use in the previous comment. Everything works from testing it a few times but I'm waiting on a response from john.

**Comment by johnsaigle** [comment]
Yeah this is in my court. I'm busy with UK biobank stuff but I'll get to this as soon as I can

**Comment by ridz1208** [comment]
@johnsaigle can you allow @maltheism to push directly to your branch or issue a PR to your fork ?

**Comment by johnsaigle** [comment]
I think anyone can push to my branches but let me know if it doesn't work

**Comment by maltheism** [comment]
Hi @ridz1208 I pushed the changes to the PR of @johnsaigle.

**Pull request merged**